### PR TITLE
EVG-2994: periodic job for new scheduler behind a feature flag

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: model/distro/db.go [hunk 1]
```diff
@@ -37,6 +38,31 @@ func FindOne(query db.Q) (Distro, error) {
 	return d, db.FindOneQ(Collection, query, &d)
 }
 
+func FindAllNames() ([]string, error) {
+	out := struct {
```
Comment (john-m-liu) [inline model/distro/db.go]: Is it possible to not have this array and have the aggregation unmarshal into a []string? Maybe with a $objectToArray in the last stage?

File: model/task_queue.go [hunk 2]
```diff
@@ -237,6 +240,33 @@ func FindAllTaskQueues() ([]TaskQueue, error) {
 	return taskQueues, err
 }
 
+func FindTaskQueueGenerationTimes() (map[string]time.Time, error) {
```
Comment (john-m-liu) [inline model/task_queue.go]: Could you add a test for this?

File: model/distro/db.go [hunk 3]
```diff
@@ -37,6 +38,31 @@ func FindOne(query db.Q) (Distro, error) {
 	return d, db.FindOneQ(Collection, query, &d)
 }
 
+func FindAllNames() ([]string, error) {
```
Comment (john-m-liu) [inline model/distro/db.go]: Could you add a test for this?

File: operations/service_runner.go [hunk 4]
```diff
@@ -29,6 +29,8 @@ import (
 	"github.com/urfave/cli"
 )
 
+const useNewScheduler = false
```
Comment (bsamek) [inline operations/service_runner.go]: Could we put this in the db?

File: operations/service_runner.go [hunk 5]
```diff
@@ -140,6 +142,7 @@ func startSystemCronJobs(ctx context.Context, env evergreen.Environment) {
 
 	const (
 		monitoringInterval      = time.Minute
+		taskPlanningInterval    = 15 * time.Second
```
Comment (bsamek) [inline operations/service_runner.go]: Shouldn't this match `20 * time.Second` on 267? Also, you could put the constant there and use it both places.

File: operations/service_runner.go (hunk 4 above)
Comment (bsamek) [reply operations/service_runner.go]: sgtm

**Pull request merged**